library/systemd/src/lib/yast2/systemd/unit.rb

```diff
@@ -0,0 +1,356 @@
+      def command(command_name, options = {})
+        command = "#{command_name} #{unit_name}.#{unit_type} #{options[:options]}"
+        Systemctl.execute(command)
+      end
+
+    private
+
+      # Run a command, pass its stderr to {#error}, {#refresh!}.
+      # @yieldreturn [#command,#stdout,#stderr,#exit]
+      # @return [Boolean] success? (exit was zero)
+      def run_command!
+        error.clear
+        command_result = yield
+        error << command_result.stderr
+        refresh!
+        command_result.exit.zero?
+      end
+
+      # Structure holding  properties of systemd unit
+      class Properties < OpenStruct
+        include Yast::Logger
+
+        # @param systemd_unit [Systemd::Unit]
+        # @param property_text [String,nil]
+        #   if provided, use it instead of calling `systemctl show`
+        def initialize(systemd_unit, property_text)
+          super()
+          self[:systemd_unit] = systemd_unit
+          if property_text.nil?
+            raw_output   = load_systemd_properties
+            self[:raw]   = raw_output.stdout
+            self[:error] = raw_output.stderr
+            self[:exit]  = raw_output.exit
+          else
+            self[:raw]   = property_text
+            self[:error] = ""
+            self[:exit]  = 0
+          end
+
+          if !exit.zero? || !error.empty?
+            message = "Failed to get properties for unit '#{systemd_unit.unit_name}' ; "
+            message << "Command `#{raw_output.command}` returned error: #{error}"
+            log.error(message)
+            self[:not_found?] = true
+            return
+          end
+
+          extract_properties
+          self[:active?]     = ACTIVE_STATES.include?(active_state)
+          self[:running?]    = sub_state    == "running"
+          self[:loaded?]     = load_state   == "loaded"
+          self[:not_found?]  = load_state   == "not-found"
+          self[:static?]     = load_state   == "static"
+          self[:enabled?]    = read_enabled_state
+          self[:supported?]  = SUPPORTED_STATES.include?(unit_file_state)
+          self[:can_reload?] = can_reload == "yes"
+        end
+
+      private
+
+        # Check the value of #unit_file_state; its value mirrors UnitFileState dbus property
+        # @return [Boolean] True if enabled, False if not
+        def read_enabled_state
+          # If UnitFileState property is missing (due to e.g. legacy sysvinit service) or
+          # has an unknown entry (e.g. "bad") we must use a different way how to get the
+          # real status of the service.
+          if SUPPORTED_STATES.include?(unit_file_state)
+            state_name_enabled?(unit_file_state)
+          else
+            # Check for exit code of `systemctl is-enabled systemd_unit.name` ; additionally
+            # test the stdout of the command for valid values when the service is enabled
+            # http://www.freedesktop.org/software/systemd/man/systemctl.html#is-enabled%20NAME...
```

Comment:
The correct link has changed to `http://www.freedesktop.org/software/systemd/man/systemctl.html#is-enabled UNIT…` (not nice anchor name IMHO).